File: library/src/main/java/renetik/android/lang/CSAsyncTask.java
package renetik.android.lang;

import android.os.AsyncTask;
import renetik.java.callback.CSRun;

public abstract class CSAsyncTask<Argument, Progress, Result> extends
		AsyncTask<Argument, Progress, Result> implements CSRun {

	public CSAsyncTask() {
		execute((Argument) null);
	}

	@Override protected Result doInBackground(Argument... params) {
		run();
		return null;
	}
}

File: library/renetik-android/src/main/java/renetik/android/lang/CSAsyncTask.kt
package renetik.android.lang

import android.os.AsyncTask

class CSAsyncTask<Result>(val run: () -> Result) : AsyncTask<Unit, Unit, Result>() {

    init {
        execute(Unit)
    }

    override fun doInBackground(vararg params: Unit): Result? = run()
}
